### PR TITLE
Add support for fewer components in input files

### PR DESCRIPTION
**Pull request by aqnuep**
Fixes issue #751.

### CODE REVIEW

**Review by MarkCallow — COMMENTED** [review]
Thank you very much for doing this. I have one question in the comment below.

File: tools/ktx/command_create.cpp [hunk 1]
```diff
@@ -1259,99 +1259,22 @@ std::unique_ptr<Image> CommandCreate::loadInputImage(ImageInput& inputImageFile)
...
-            image = std::make_unique<rgb32fimage>(width, height);
-            loadFormat = createFormatDescriptor(VK_FORMAT_R32G32B32_SFLOAT, *this);
-            break;
-        case 4:
-            image = std::make_unique<rgba32fimage>(width, height);
-            loadFormat = createFormatDescriptor(VK_FORMAT_R32G32B32A32_SFLOAT, *this);
-            break;
-        }
+        image = std::make_unique<rgba32fimage>(width, height);
+        loadFormat = createFormatDescriptor(VK_FORMAT_R32G32B32A32_SFLOAT, *this);
```
[the remark this answers is not in the document]
Comment (aqnuep) [reply tools/ktx/command_create.cpp:1266]: Before, we had the following cases for PNG:

1. Luminance images were loaded as RGB
2. Luminance-alpha images were loaded as RGBA
3. RGB images were loaded as RGB
4. RGBA images were loaded as RGBA

So as we want to be able to use RGB as RGBA, on the PNG side it made sense to just always use 4 components. There seems to be little reason to try to optimize for the case when RGB will indeed be used as RGB.

I followed the same principle on the EXR side, for the same reason, as with this requirement we want to be able to potentially create RGBA textures out of e.g. R-only images.

Sure, things could be potentially optimized, but despite our improvements to the image.hpp library, it's still not a fully flexible image manipulation library that is able to support all operations with inputs and outputs of any type and any component count.

We can think about changing that, adding further capabilities and flexibility to image.hpp, but that would require a wider sweep over the codebase. This seemed to be the best solution to enable the sort of use cases requested in issue #751 without wider effects.

[the remark this answers is not in the document]
Comment (MarkCallow) [reply tools/ktx/command_create.cpp:1266]: Thanks for the explanation. It is much as I thought but I wanted to confirm.

**Comment by MarkCallow** [comment]
@aqnuep please fix the conflict that has just arisen. Likely it is from PR #749. Sorry for the extra work.

**Comment by aqnuep** [comment]
@MarkCallow, if you didn't notice, this one is rebased and ready to go.

**Pull request merged**